lib/rubocop/cop/rspec/factory_bot/factory_class_name.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      module FactoryBot
+        # Use string value when setting the class attribute explicitly.
+        #
+        # @example
+        #   # bad
+        #   factory :foo, class: Foo do
+        #   end
+        #
+        #   # good
+        #   factory :foo, class: 'Foo' do
+        #   end
+        class FactoryClassName < Cop
+          MSG = "Pass '%<class_name>s' instead of %<class_name>s."
+
+          def_node_matcher :factory_call?, <<~PATTERN
+            (block (send _ :factory ...) (...) _)
```

Comment:
I suggest using [children matching](https://github.com/rubocop-hq/rubocop/issues/6965) to find `class` key, you can find some examples [here](https://github.com/rubocop-hq/rubocop-rspec/pull/804).